docs/content/docs/how-to/add-handwritten-datasource-documentation.md

```diff
@@ -0,0 +1,157 @@
+
+```
+mmv1/third_party/terraform/website/docs/
+├─ guides/
+│  ├─ ...
+├─ d/
+│  ├─ ...
+├─ r/
+│  ├─ ...
+├─ index.html.markdown
+```
+
+The subfolder `d` corresponds to data sources, and inside there is a file for each data source.
+
+## Creating the new markdown file
+
+Next you need to add the file for the new data source's documentation. Create a new file inside the `d` folder. The filename should be the name of the data source with the `google_` prefix removed. For example if you were adding a new data source `google_foobar` then you would need to create a new file with the name `foobar.html.markdown`.
+
+## Adding contents
+
+Pages within the documentation need to be consistent and contain the sections that users expect. Below is some guidance about the different sections to include and what their contents should be.
+
+### Frontmatter
+
+The top of the file needs to contain frontmatter, which is used to create the new documentation page's title and manage how the page is linked to in the documentation's sidebar navigation. It it not rendered but it is very important for the new page in the documentation to be generated and available to users.
+
+You need to make sure your file's frontmatter includes:
+- `subcategory` - This sets which section in the sidebar navigation (left side) the new page is categorised into.
+- `page_title` - This frontmatter is specific to files in the Guides section (`/website/docs/guides`) but we set it for all markdown files.
+- `description` - A decription of the page.
+
+For example, here is the frontmatter from `/website/docs/d/container_cluster.html.markdown` ([link to generated page](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/container_cluster)):
+
+```markdown
+---
+subcategory: "Kubernetes (Container) Engine"
+page_title: "Google: google_container_cluster"
+description: |-
+  Get info about a Google Kubernetes Engine cluster.
+---
+```
+
+### Page title and description
+
+The next section in the markdown file is rendered as the first part of the page body.
+
+It should contain:
+- the page title, as an H1 header
+- a description general information about the data source
+
+The description can be as long or as short as necessary. The minimum information that's included in this section are links to official documentation and the API reference pages. Other guidance, warnings, or explanations of concepts can be included here. To create pronounced warning or info sections, see the [this documentation](/magic-modules/docs/getting-started/provider-documentation/#what-formatting-is-available) page for info.
+
+For example, here's the title and opening description of `/website/docs/d/kms_crypto_key.html.markdown` ([link to generated page](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/kms_crypto_key)):
+
+```markdown
+# google\_kms\_crypto\_key
+
+Provides access to a Google Cloud Platform KMS CryptoKey. For more information see
```

Comment:
I found a handwritten example that mimics the generated docs format, but the majority of handwritten docs vary a lot in the description area (and typically for the better - lots of extra info and links).

Reply:
Yeah, we're adding less extra info and links over time. I'd like to make the two common links (user guide and API reference required in MMv1 at some point, with an explicit opt-out when not available (but it would also be pretty annoying to need to add them to iterate locally)